www/js/splash/pushNotifySettings.ts

```diff
@@ -0,0 +1,241 @@
+};
+
+/**
+ * @function registers for notifications and updates user
+ * currently called on reconsent and on intro done
+ */
+const registerPush = function () {
+  registerPromise()
+    .then(function (t) {
+      // alert("Token = "+JSON.stringify(t));
+      logDebug('Token = ' + JSON.stringify(t));
+      return window['cordova'].plugins.BEMServerSync.getConfig()
+        .then(
+          function (config) {
+            return config.sync_interval;
+          },
+          function (error) {
+            console.log('Got error ' + error + ' while reading config, returning default = 3600');
+            return 3600;
+          },
+        )
+        .then(function (sync_interval) {
+          updateUser({
+            device_token: t['token'],
+            curr_platform: window['cordova'].platformId,
+            curr_sync_interval: sync_interval,
+          });
+          return t;
+        });
+    })
+    .then(function (t) {
+      // alert("Finished saving token = "+JSON.stringify(t.token));
+      logDebug('Finished saving token = ' + JSON.stringify(t.token));
+    })
+    .catch(function (error) {
+      displayError(error, 'Error in registering push notifications');
+    });
+};
+
+/**
+ * @function handles silent push notifications
+ * works with BEMDataCollection plugin
+ * @param data from the notification
+ * @returns early if platform is not ios
+ */
+const redirectSilentPush = function (event, data) {
+  logDebug('Found silent push notification, for platform ' + window['cordova'].platformId);
+  if (window['cordova'].platformId != 'ios') {
+    logDebug('Platform is not ios, handleSilentPush is not implemented or needed');
+    // doesn't matter if we finish or not because platforms other than ios don't care
+    return;
+  }
+  logDebug('Platform is ios, calling handleSilentPush on DataCollection');
+  var notId = data.additionalData.payload.notId;
+  var finishErrFn = function (error) {
+    logDebug('in push.finish, error = ' + error);
+  };
+
+  window['cordova'].plugins.BEMDataCollection.getConfig()
+    .then(function (config) {
+      if (config.ios_use_remote_push_for_sync) {
+        window['cordova'].plugins.BEMDataCollection.handleSilentPush().then(function () {
+          logDebug('silent push finished successfully, calling push.finish');
+          showDebugLocalNotification('silent push finished, calling push.finish');
+          push.finish(function () {}, finishErrFn, notId);
+        });
+      } else {
+        logDebug('Using background fetch for sync, no need to redirect push');
+        push.finish(function () {}, finishErrFn, notId);
+      }
+    })
+    .catch(function (error) {
+      push.finish(function () {}, finishErrFn, notId);
+      displayError(error, 'Error while redirecting silent push');
+    });
+};
+
+/**
+ * @function shows debug notifications if simulating user interaction
+ * @param message string to display in the degug notif
+ */
+var showDebugLocalNotification = function (message) {
```

Comment:
`const` here

(or simply `function showDebugLocalNotification(message) {...}`)

Reply:
Done in [85d2884](https://github.com/e-mission/e-mission-phone/pull/1093/commits/85d2884128aecc351ccaafd12a7ab553f8385e83) -- I decided to keep the const to stay consistent with the rest of the file